yarn-project/archiver/src/archiver/data_retrieval.ts

```diff
@@ -122,3 +124,33 @@ export async function retrieveNewContractData(
   } while (blockUntilSynced && searchStartBlock <= currentBlockNumber);
   return { nextEthBlockNumber: searchStartBlock, retrievedData: retrievedNewContracts };
 }
+
+/**
+ * Ftech new L1 to L2 messages.
+ * @param publicClient - The viem public client to use for transaction retrieval.
+ * @param inboxAddress - The address of the inbox contract to fetch messages from.
+ * @param blockUntilSynced - If true, blocks until the archiver has fully synced.
+ * @param currentBlockNumber - Latest available block number in the ETH node.
+ * @param searchStartBlock - The block number to use for starting the search.
+ * @returns An array of L1ToL2Message and next eth block to search from.
+ */
+export async function retrieveNewPendingL1ToL2Messages(
+  publicClient: PublicClient,
+  inboxAddress: EthAddress,
+  blockUntilSynced: boolean,
+  currentBlockNumber: bigint,
+  searchStartBlock: bigint,
+) {
```

Comment:
Add a type for the return value.

Reply:
Ah I tried but it gets quite ugly (which is why none of the methods here have a return type.
Basically you add a return type like `: Promise<{ nextEthBlockNumber: bigint; retrievedData: L1ToL2Message[]}>` and then the linter says `Missing JDoc statement` - except the linter wants the jdoc statement in the return statement itself (L143) as opposed to on L135.

Reply:
Could you make a generic type for each:
```ts
/**
 * Data retreived from logs
 */
type RollupLog<T> = {
  /**
   * The next block number.
   */
  nextEthBlockNumber: bigint;
  /**
   * The data returned.
   */
  retrievedData: T[];
}

```

Then have the return type be:
```ts
func something(): Promise<RollupLog<L1ToL2Message>> {}
```

should do the trick